### PR TITLE
Remove laserturret

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Remove laserturret"

#### Purpose of change
Solar powered laser turrets are absurd.

#### Describe the solution
Remove laserturret.

#### Testing
Game loads correctly.

### CODE REVIEW

**Comment by ussdefiant60** [comment]
Two questions come to mind when i look at this:

1) Where do they spawn currently? Places that conceiveably could have a fixed power supply set up for them? Maybe a fluff adjustment rather than outright removal?

2)ATM, if i'm reading JSONs right, the only source of laser cannons in the game is by disassembling these guys either in their broken or undeployed forms. Should perhaps another method be added to acquire them for top-end deathmobile builders even if turrets go away?

**Comment by anothersimulacrum** [comment]
They don't spawn anywhere.

If there's a sensible place that there'd be that sort of thing (we already have some laser weapons, these are somewhat redundant with those), yes, but I don't think we do, and as such, they should be obsoleted with these. If it turns out there, is, we can fix that later.

**Comment by Zireael07** [comment]
Maybe replace some of the lab turrets with those (and refluff them to be fixed power, not solar) ?
I seem to remember a lab variant that's even HIGHER tech than usual?

File: data/json/items/obsolete.json [hunk 1]
```diff
@@ -2717,5 +2717,88 @@
...
+    "material": [ "steel", "plastic" ],
+    "symbol": ";",
+    "color": "white",
+    "use_action": {
+      "type": "place_monster",
+      "monster_id": "mon_laserturret",
+      "difficulty": 6,
+      "moves": 100,
+      "skill1": "electronics",
+      "skill2": "computer"
```
Comment (kevingranade) [inline data/json/items/obsolete.json:2741]:
```suggestion
       "skills": [ "electronics", "computer" ]
```

File: data/json/items/obsolete.json [hunk 2]
```diff
@@ -2717,5 +2717,87 @@
...
+    "bashing": 8,
+    "material": [ "steel", "plastic" ],
+    "symbol": ";",
+    "color": "white",
+    "use_action": {
+      "type": "place_monster",
+      "monster_id": "mon_laserturret",
+      "difficulty": 6,
+      "moves": 100,
+       "skills": [ "electronics", "computer" ]
```
Comment (ZhilkinSerg) [inline data/json/items/obsolete.json:2740]:
```suggestion
      "skills": [ "electronics", "computer" ]
```

**Comment by NoahGoldFox** [comment]
Calling something "absurb" is not a reason at all to remove it! Remember "improve, don't remove". Removing laser-turrets is just going to make the game less fun and interesting!

**Comment by ZhilkinSerg** [comment]
> 
> 
> Calling something "absurb" is not a reason at all to remove it! Remember "improve, don't remove". Removing laser-turrets is just going to make the game less fun and interesting!

Solar powered turrets are nothing, but absurd.

**Comment by anothersimulacrum** [comment]
Laser turrets were already practically removed, this is just removing mostly dead data.

> Remember "improve, don't remove". 

Removal is improving - if something makes sense, it's better to rip it out until someone wants to implement it in a way that makes sense than leave it in.
Not having it is an improvement over having it.

**Comment by NoahGoldFox** [comment]
> Removal is improving - if something makes sense, it's better to rip it out until someone wants to implement it in a way that makes sense than leave it in.
> Not having it is an improvement over having it.

But thats not what happens, most of the time nobody does think of a way to bring them back and the content is gone forever.

**Comment by anothersimulacrum** [comment]
Okay? The game is better than it was when the content existed, so it's not a loss.

**Comment by NoahGoldFox** [comment]
> Okay? The game is better than it was when the content existed, so it's not a loss.

It is not better. Its another fun weapon option removed just for small scientific reasons

**Comment by anothersimulacrum** [comment]
It makes the universe make more sense and be more coherent, that's an improvement.

**Comment by Kenan2000** [comment]
Can this be moved to Aftershock at least?It would make sense

**Comment by Axema** [comment]
Yeah i definitely agree, considering that this is a sci-fi turret that despite being unrealistic is capable of existing in a sci-fi version of cataclysm, it would be a wise decision to add it to Aftershock.

**Comment by EvgenijM86** [comment]
A solar powered laser turret would actually make sense in cataclysm setting, but not as a weapon to kill things, but rather as a tool to blind whatever is pursuing you car. There is nothing scifi about automatic face recognition module that would detect anything that looks like an eye and point a weak laser turret directly at it.

**Comment by Zireael07** [comment]
Solar is absurd, but how difficult would it be to refluff those to fixed power and put them in the high-tech lab variant?

**Comment by Stuntaneous** [comment]
Solar power + capacitors. Not absurd.

**Comment by ZhilkinSerg** [comment]
> 
> 
> Solar power + capacitors. Not absurd.

Still absurd.

**Comment by GGgatherer** [comment]
What is actually absurd is solar panels' power output compared to laser turret's power requirements. But I get it, it's easier to outright remove one part of equation therefore removing the equation itself, than try finding out what's wrong with it.

**Comment by morthein** [comment]
Can we make it so the laser powered turret draws a huge charge per shot from the battery? It won't be able to shoot a lot unless you have a ton of stored energy.
That way we can have both --  

*Don't forget about the word nerf!*

**Comment by yeebusters** [comment]
I don't think I've ever seen a laser turret in all my time playing the game...didn't even know they existed

**Comment by l2evy** [comment]
I have a different playstyle than most people I think, maybe my input may be of some use.

I tend to use classic zombies and "no scifi" and "no fictional guns" , i prefer my zombie apoc to be romero type of slow but overwhelming in high spawn rate and realistic guns and items.

Having said all that, I can't just see this as a win. I know my chosen play style is probably not mainstream so I get why some people might get a little frustrated by this change. Removing something just because  "absurd" is sorta vague reasoning. I can think of a lot of things in this game that many would consider absurd but in context aimed at the player disadvantage instead of an advantage. I think if I played with everything enabled default, all the nether creatures, zombie bosses following the default point allocations for my character, I would eventually want some kind of stationary laser turret, even if it took a huge amount of fuel/power to charge the caps for a only a few burst for base defense. So I would feel a bit more safe with a panic button to at least help some before I am writing my last words on a tombstone.

I would say no easy powered options like a few cheap solar panels ripped off a car and you have a death laser to melt anything in the game. Higher power laser = more power required vastly with capacitors or battery banks, making the unit most likely stationary (which a mobile death laser would be a bit absurd) but I do like the idea of maybe leaving a weaker form in that could be mobile, maybe something to burn the flesh a little of weaker nether creatures maybe make them flee or teleport away, maybe some things enrage and only get stronger with more desire to smash you into the dirt. Having a death  laser still available is absurd? Well think of it in a time aspect if they are not mobile, that means you have to bring the parts and build it where you want it, which if set correctly would take a long time to setup, then you still have to charge it. In the time you are doing that, you could have been just killing whatever you want to kill with the laser, with some other form of weapon that's easy, such as laser rifles or something. All that considered it then becomes a option instead of a OP endgame thing. Spend a whole day setting up and charging a laser to help clear some hard mobs in town? or just go in there as usual and clear it yourself with some kiting and choice handheld weaponry to handle the job.

**Comment by TechyBen** [comment]
> Solar powered laser turrets are nothing, but absurd.

Yes. Then that is a power conversion problem. Is the power draw rate of the Laser turret not easy to mod?

Plus, do we not then just reduce the output too? So it's a burning laser, not a plasma inducing arch reactor.

Two factors easy to adjust without removing them completely.

**Comment by TechyBen** [comment]
> Solar power + capacitors. Not absurd.

Don't even need that. Close range metal melting/forging can be done with a mirrors a bit bigger than a small car. Granted, it needs to be full sun/a desert, but there are youtube video of people melting glass/metal with concave mirrors/convex lenses.

**Comment by morthein** [comment]
You could just move it to the day sky's above mod (I think that's what it's called, not at a pc right now) isn't that focused more on aliens and sci fi?

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by anothersimulacrum** [comment]
Things are moved to mods because mod maintainers want them. If the maintainers of Aftershock or Dark Skies Above want them, they will add it. I'm interpreting their silence on the matter to mean they don't.

**Comment by dkavlakov** [comment]
I remember, once upon a time, in early 1980's a solar powered flashlight with nickel-cadmium battery and tungsten wire lamp bubble was placed on a science magazine's front page as "example of completely absurd invention". Now-days solar powered flashlights with lithium-polymer batteries and led lights are just another standard consumer product and no one calls them "absurd". It is just matter of knowledge, usability and efficiency. And in this context - green semi-conductive lasers with current technology have around 30% power efficiency, but theoretically can reach over 80%.

**Comment by ZhilkinSerg** [comment]
> 
> 
> I remember, once upon a time, in early 1980's a solar powered flashlight with nickel-cadmium battery and tungsten wire lamp bubble was placed on a science magazine's cover page as "example of completely absurd and useless invention". Now-days solar powered flashlights with amorphous silicon solar cells, lithium-polymer batteries and led lights are just another standard consumer product and no one calls them "absurd" or "useless". It is just matter of knowledge, usability and efficiency. And in this context - green semi-conductive lasers with current technology have less than 30% power efficiency, but theoretically can reach over 80%.

How is it helping to dismiss "solar powered lasers are absurd"?

**Comment by skdursh** [comment]
"Absurd" or not, personal opinion is not the be-all-end-all of the matter and if you look at pretty much every comment on here in regards to this they near unanimously disagree with removal.

**Comment by anothersimulacrum** [comment]
[This isn't a democracy](https://github.com/CleverRaven/Cataclysm-DDA/wiki/Frequently-Made-Suggestions#Project_Management).

**Comment by dkavlakov** [comment]
> How is it helping to dismiss "solar powered lasers are absurd"?

Depending how exactly do you consider them to be absurd:

- From technological point: even now they are inadequate, not absurd.
- From political/religious/ideological point of view: state already regulates them, but this does not and will not prevent military and business use as long as they are technologically adequate.
- From gameplay point of view: If above two are OK then what is wrong is either the gameplay itself, or your concept of it (BTW: Good example of fixing what is OK due to weird gameplay concepts was the recent nerf of bows).

**Comment by ZhilkinSerg** [comment]
> 
> 
> > How is it helping to dismiss "solar powered lasers are absurd"?
> 
> Depending how exactly do you consider them to be absurd:
> 
>     * From technological point: even now they are inadequate, not absurd.
> 
>     * From political/religious/ideological point of view: state already regulates them, but this does not and will not prevent military and business use as long as they are technologically adequate.
> 
>     * From gameplay point of view: If above two are OK then what is wrong is either the gameplay itself, or your concept of it (BTW: Good example of fixing what is OK due to weird gameplay concepts was the recent nerf of bows).

Solar powered lasers are absurd

**Comment by kevingranade** [comment]
I don't follow, the flashlight in question WAS an absurd and useless invention, it required substantial overhauls to essentially every component before it stopped being useless, and those overhauls weren't even *hypothetical* possibilities at the point when the magazine was published.  It's a very on-point analogy, but it says the opposite of what you are saying it does.  If a device is laughably unfit for purpose given the current level of technology or even the current understanding of physical constraints of the system, that means it's an absurd and pointless device.  It doesn't matter that future advancements *might* make something resembling it fit for purpose, the device itself that is not fit for purpose is absurd.

Some napkin math says that a self contained solar powered antipersonnel laser turret based on modern technology would have a duty cycle approaching a handful of potentially lethal shots per day, and be incredibly fragile compared to more conventional designs because of the inherent fragility and size of the solar panels.  It would almost certainly be many times more expensive than a competing turret system based on a conventional firearm as well.  That outlines a device that doesn't remotely approach being fit for purpose without a complete overhaul of every major technology it is based on.  That is a useless and absurd device.

**Comment by dkavlakov** [comment]
I agree that such laser turret's capacity with current technology is ~2-3 shots/24h, but with unmanned lifespan that can reach over 2 years. This is pretty suitable for certain applications. As for the fragility - amorphous silicon solar cells use polymer base in civilian applications, but for military their usual base is steel plate - something that takes much less damage from. Also damage decreases output of such cells, but does not disable them. Similar is the situation with lithium-ceramic batteries, that are currently in pre-production stage and will definitely become choice for military applications due to their ability to take immense damage, but still give output.

This is with current level of technology. C:DDA however is placed ~30 years in the future and already assumes significant (actually even questionable) level of technological advancement during them.

**Comment by anothersimulacrum** [comment]
> C:DDA however is placed ~30 years in the future

[It's not](https://cataclysmdda.org/design-doc/).

**Comment by dkavlakov** [comment]
> > C:DDA however is placed ~30 years in the future
> 
> [It's not](https://cataclysmdda.org/lore-background.html).

There is no exact info on when year -10 is, but while most date references are not exact they give good average. For example 9x19 and .45 ammunition was considered to be 150 years old during the Catalysm. This places year 0 at least somewhere around 2053-2054.

**Comment by Zireael07** [comment]
@dkavlakov: This is a remnant of the period when the game was set circa 2040. It no longer is the case, as @anothersimulacrum pointed out.

**Comment by anothersimulacrum** [comment]
image
The 9mm and .45 ammo had their descriptions corrected, but this appears to have been accidentally reverted in 96b1c34e40863680c82958e5c90345d73901ef29, part of #33302, but it doesn't appear in the changes there? Odd.
I'll fix that.

**Comment by ZhilkinSerg** [comment]
There is a revert for that:

https://github.com/CleverRaven/Cataclysm-DDA/pull/33302/commits/1b9c6e5d482e1cb38fbb9ae082e421447393f90a

And then commits were squashed, so there are no changes for that file in the final commit.

**Comment by dkavlakov** [comment]
I see. In this case however solar powered laser turrets are far less absurd than many other things in the game. Even Rivtech weapons are pure un-scientific fiction compared to them. Ceramic caseless suicide devices as standard issue weapons in US army... I bet many people in the real world dream to see such thing.

**Comment by anothersimulacrum** [comment]
One thing being wrong isn't a an argument to have another thing be wrong.

**Comment by skdursh** [comment]
I don't get what makes it "wrong" though. This is science-fiction, emphasis on the fiction.

**Comment by ZhilkinSerg** [comment]
> 
> 
> I don't get what makes it "wrong" though. This is science-fiction, emphasis on the fiction.

Yes, there is a fictional world and it has its rules. Some of these rules are not different from the rules in our real world and some are different.

**Comment by EvgenijM86** [comment]
If I add exactly the same weapon in a pull request, but rename its description to "A strange mi-go artifact, uses sunlight to charge itself, fires beams of realism" - would it be accepted?

**Comment by kevingranade** [comment]
No

**Comment by Zireael07** [comment]
Would a fixed power (not solar) variant be accepted into a mod or mainline?

**Comment by Night-Pryanik** [comment]
> Would a fixed power (not solar) variant be accepted into a mod or mainline?

Yep. It just need to be carefully located in appropriate places, like an experimental weapons lab or some military base.

**Comment by Zireael07** [comment]
Gotcha. I admit I am a bit out of the loop when it comes to the mapgen after the big rework, would placing them require code changes or is it a JSON-only job (I know the turret change itself would be JSON only)?

**Comment by Night-Pryanik** [comment]
I think json-only-based changes would be enough for first approach.

**Pull request merged**